platform/platform-impl/src/com/intellij/openapi/vfs/newvfs/RefreshSessionImpl.java

```diff
@@ -154,9 +164,13 @@ else if (!nvf.isDirty()) {
       }
       while (!myCancelled && myIsRecursive && count < 3 && ContainerUtil.exists(workQueue, f -> ((NewVirtualFile)f).isDirty()));
 
-      if (t != 0) {
-        t = System.currentTimeMillis() - t;
-        LOG.trace((myCancelled ? "cancelled, " : "done, ") + t + " ms, events " + myEvents);
+      t = System.currentTimeMillis() - t;
+      if (LOG.isTraceEnabled()) {
+        LOG.trace((myCancelled ?  "cancelled, " : "done, ") + t + " ms, events " + myEvents);
+      } else if (t > REFRESH_SESSION_DURATION_REPORT_THRESHOLD_SECONDS * 1000L){
```

Comment:
int is enough here I suppose, can flip to have it long if reqd but readability will suffer